flink-connectors/flink-connector-kafka-0.11/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducer011.java

```diff
@@ -608,6 +610,16 @@ public void onCompletion(RecordMetadata metadata, Exception exception) {
 
 		super.open(configuration);
 	}
+	private void serializeOperatorTimestampsIntoByteBuffer(List<OperatorOutputTimestamp> operatorOutputTimestamps, ByteBuffer target) {
+		boolean first = true;
+		for(OperatorOutputTimestamp o : operatorOutputTimestamps){
+			if(first) first = false ; else target.putChar(',');
```

Comment:
Although you save lines with an if oneliner, it's not as readable.
Also mind the space between if and parenthesis, which should be there but is not. Mind similar cases, for, etc.